lib/src/request/ResponseTypeMap.ts

```diff
@@ -0,0 +1,16 @@
+import type { XmlNode } from '../xml/XmlNode.js';
+
+/**
+ * The a request's `responseType` to the corresponding data type.
+ *
+ * @group Request
+ *
+ * @beta
+ */
+export type ResponseTypeMap<T extends string | undefined> =
+	T extends 'json' ? any :
+	T extends 'text' ? string :
+	T extends 'blob' ? Blob :
+	T extends 'arraybuffer' ? ArrayBuffer :
+	T extends 'document' ? XmlNode :
```

Comment:
We are using the simpler `XmlNode` type here instead of [`XMLDocument`](https://developer.mozilla.org/en-US/docs/Web/API/XMLDocument) so that CML's `parseXml` function can also be used.